src/include/txn_inline.h

```diff
@@ -692,21 +708,57 @@ __wt_txn_visible_all(WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t times
     return (pinned_ts != WT_TS_NONE && timestamp <= pinned_ts);
 }
 
+/*
+ * __wt_txn_upd_get_durable --
+ *     Safely retrieve a durable timestamp from an update.
+ */
+static inline int
+__wt_txn_upd_get_durable(WT_SESSION_IMPL *session, WT_UPDATE *upd, wt_timestamp_t *durable_ts)
+{
+    uint8_t prepare_state;
+
+    WT_ORDERED_READ(prepare_state, upd->prepare_state);
+
+    WT_ASSERT_ALWAYS(session,
+      prepare_state == WT_PREPARE_LOCKED || prepare_state == WT_PREPARE_INPROGRESS,
+      "Trying to read durable_ts in an invalid prepare state.");
+
+    *durable_ts = upd->__durable_ts;
+
+    return (0);
+}
+
+/*
+ * __wt_txn_upd_set_durable --
+ *     Safely retrieve a durable timestamp from an update.
+ */
+static inline int
+__wt_txn_upd_set_durable(wt_timestamp_t *durable_ts, wt_timestamp_t *new_durable_ts)
```

Comment:
This should take a WT_UPDATE *, and non pointer type for the durable timestamp.

`__wt_txn_upd_set_durable(WT_UPDATE *, wt_timestamp_t new_durable_ts)`.